internal/impl/devbox.go

```diff
@@ -223,6 +225,8 @@ func (d *Devbox) RunScript(cmdName string, cmdArgs []string) error {
 	if err != nil {
 		return err
 	}
+	// Used to determine whether we're inside a shell (e.g. to prevent shell inception)
+	env["DEVBOX_SHELL_ENABLED"] = "1"
```

Comment:
nit, add comment that this is temporary because services needs it but should probably go away (because run != devbox shell)